src/screens/LoginScreen.tsx

```diff
@@ -4,52 +4,94 @@ import {
   TextInput,
   Text,
   View,
+  Linking,
 } from "react-native";
 import { styles } from "../styles";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import { atom, useAtom } from "jotai";
-import { apiKeyAtom, serverAtom } from "../atoms";
+import {
+  lnbitsUrlAtom,
+  domainAtom,
+  userAtom,
+  walletAtom,
+  adminKeyAtom,
+} from "../atoms";
+import URLParse from "url-parse";
+import axios from "axios";
 
 function Login(): JSX.Element {
-  const [apiKey, setApiKey] = useAtom(apiKeyAtom);
-  const [temporaryApiKey, setTemporaryApiKey] = useState(apiKey);
+  // Preparation for parsing the server url
+  const [lnbitsUrl, setLnbitsUrl] = useAtom(lnbitsUrlAtom);
+  const [domain, setDomain] = useAtom(domainAtom);
+  const [user, setUser] = useAtom(userAtom);
+  const [wallet, setWallet] = useAtom(walletAtom);
+  const [adminKey, setAdminKey] = useAtom(adminKeyAtom);
+
+  useEffect(() => {
+    const parsedData = parseUrl(lnbitsUrl);
+
+    setDomain(parsedData.domain || "");
+    setUser(parsedData.user || "");
+    setWallet(parsedData.wallet || "");
+  }, [lnbitsUrl]);
+
+  function parseUrl(url: string) {
+    const fullUrl = new URLParse(url, true);
+
+    const domain = fullUrl.origin;
+    const user = fullUrl.query.usr;
+    const wallet = fullUrl.query.wal;
+
+    return { domain, user, wallet };
+  }
+
+  async function fetchAdminKey(lnbitsUrl: string) {
+    const response = await axios.get(lnbitsUrl);
+    const parsedApiKey = response.data.match(
+      /<strong>Admin key: <\/strong><em>([\da-fA-F]{32})<\/em><br \/>/
+    )[1];
+
+    setAdminKey(parsedApiKey);
+  }
 
-  const defaultServer = (): string => {
-    if (server) {
-      return server;
-    } else return "https://legend.lnbits.com/";
+  const handleOpenWallet = () => {
+    Linking.openURL(lnbitsUrl);
   };
-  const [server, setServer] = useAtom(serverAtom);
-  const [temporaryServer, setTemporaryServer] = useState(defaultServer);
 
-  const handleButtonClick = () => {
-    setApiKey(temporaryApiKey);
-    setServer(temporaryServer);
+  const handleButtonClick = async () => {
+    try {
+      await fetchAdminKey(lnbitsUrl);
+
+      const parsedData = parseUrl(lnbitsUrl);
+
+      setDomain(parsedData.domain || "");
+      setUser(parsedData.user || "");
+      setWallet(parsedData.wallet || "");
+    } catch (error) {
+      console.log(error);
+    }
   };
 
   return (
     <View style={styles.container}>
       <SafeAreaView>
-        <Text>Your LNbits apikey:</Text>
+        <Text>LNbits URL:</Text>
         <TextInput
           style={styles.input}
-          onChangeText={setTemporaryApiKey}
-          placeholder={temporaryApiKey}
+          onChangeText={setLnbitsUrl}
+          placeholder={lnbitsUrl}
```

Comment:
Placeholder is a value, that is displayed usually in gray color, when there is no text in the input. Use `value` instead.

```suggestion
          value={lnbitsUrl}
```